cirq-google/cirq_google/json_resolver_cache.py

```diff
@@ -20,11 +21,19 @@
 
 @functools.lru_cache()
 def _class_resolver_dictionary() -> Dict[str, ObjectFactory]:
+    def _old_xmon(*args, **kwargs):
+        warnings.warn(
+            'Attempted to json load a Bristlecone or Foxtail Device.'
+            'These devices were removed in Cirq v0.15 and are no '
+            'longer supported. Please update device usage to a '
+            'supported device or downgrade your Cirq installation.'
+        )
+        return None
```

Comment:
I was suggesting we return the name or something. I think if we're at the point of returning None, we should just be raising an exception. Also: the error message can tell you what the name of the device is that failed to load.

Reply:
Updated to return the name and added more info to the warning.